connectors/sources/confluence.py

```diff
@@ -333,6 +372,16 @@ async def fetch_server_space_permission(self, url):
                 f"Something went wrong. Make sure you have installed Extender for running confluence datacenter/server DLS. Exception: {exception}."
             )
             return {}
+        except ServerDisconnectedError:
+            raise
+        except ClientPayloadError as exception:
+            self._logger.error(
+                f"Payload error occurred while fetching {url}. Exception: {exception}."
+            )
+        except Exception as exception:
+            self._logger.exception(
+                f"Something went wrong while fetching from {url}. Exception: {exception}."
+            )
```

Comment:
You've added this block in a lot of places. This is just swallowing _more_ errors that would currently have bubbled up. Why?

Reply:
I think you're misunderstanding my question. The issue this PR is attached to was filed because the connector wasn't raising enough errors. Instead of surfacing them, this _swallows_ more errors. The jobs would complete successfully, and you'd need to be monitoring the logs to see that there were issues. That's not a good user experience if there are significant problems occurring.

Reply:
In the current implementation, we raise exceptions specifically for ServerDisconnectError and ClientResponseError if not handled, while for any other encountered exceptions, we log the error and increment the `api_failed_count`.

Reply:
Yes, I understand that's what the code does in this branch. I'm saying that's probably not solving the issue raised in the linked ticket. 
If there's some error that isn't a ServerDisconnectError and isn't a ClientResponseError, by not re-raising it you don't let the caller determine how problematic the failure is. And if a critical request fails, but just that _one_ critical request, where a lot of less important requests succeed, the error threshold may not be high enough for the sync to register as a failure.